sdks/java/core/src/main/java/org/apache/beam/sdk/util/BigQueryServices.java

```diff
@@ -67,4 +83,51 @@ void startQueryJob(String jobId, JobConfigurationQuery query, boolean dryRun)
     Job pollJob(String projectId, String jobId, int maxAttempts)
         throws InterruptedException, IOException;
   }
+
+  /**
+   * An interface for the Cloud BigQuery table service.
+   */
+  public interface TableService {
+    /**
+     * Gets the specified table resource by table ID.
+     */
+    Table getTable(String projectId, String datasetId, String tableId)
+        throws InterruptedException, IOException;
+
+    Dataset createDataset(String projectId, String datasetId, String location, String description)
+        throws IOException, InterruptedException;
+  }
+
+  /**
+   * An interface to read the Cloud BigQuery directly.
```

Comment:
not sure why we need to refer to BoundedSource.
Even though now it is the only use case, this interface doesn't tie to BoundedSource.